x/mint/keeper/mint_test.go

```diff
@@ -0,0 +1,212 @@
+	stakingKeeper *minttestutil.MockStakingKeeper
+	bankKeeper    *minttestutil.MockBankKeeper
+}
+
+// TestMintFnTestSuite runs the mint test suite.
+func TestMintFnTestSuite(t *testing.T) {
+	suite.Run(t, new(MintFnTestSuite))
+}
+
+// SetupTest sets up the context, KV store, and mocks.
+func (s *MintFnTestSuite) SetupTest() {
+	encCfg := moduletestutil.MakeTestEncodingConfig(mint.AppModuleBasic{})
+	key := storetypes.NewKVStoreKey(types.StoreKey)
+	storeService := runtime.NewKVStoreService(key)
+	testCtx := testutil.DefaultContextWithDB(s.T(), key, storetypes.NewTransientStoreKey("transient_test"))
+	s.ctx = testCtx.Ctx
+
+	ctrl := gomock.NewController(s.T())
+	accountKeeper := minttestutil.NewMockAccountKeeper(ctrl)
+	s.bankKeeper = minttestutil.NewMockBankKeeper(ctrl)
+	s.stakingKeeper = minttestutil.NewMockStakingKeeper(ctrl)
+
+	// Return a dummy module address for the mint module.
+	accountKeeper.EXPECT().GetModuleAddress(types.ModuleName).Return(sdk.AccAddress{}).AnyTimes()
+
+	// Override the default mint function with our dummy inflation calculator.
+	s.mintKeeper = keeper.NewKeeper(
+		encCfg.Codec,
+		storeService,
+		s.stakingKeeper,
+		accountKeeper,
+		s.bankKeeper,
+		authtypes.FeeCollectorName,
+		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		// keeper.WithMintFn(CUSTOM MINT FN HERE),
+	)
+
+	// Set default parameters.
+	err := s.mintKeeper.Params.Set(s.ctx, types.DefaultParams())
+	s.Require().NoError(err)
+
+	// Set a known dummy minter in the store for deterministic behavior.
+	s.Require().NoError(s.mintKeeper.Minter.Set(s.ctx, types.DefaultInitialMinter()))
+}
+
+// TestDefaultMintFn_Success tests the successful execution of the default mint function.
+func (s *MintFnTestSuite) TestDefaultMintFn_Success() {
+	// Set the staking keeper expectations.
+	stakingSupply := math.NewInt(1_000_000_000)
+	bondedRatio := math.LegacyNewDecWithPrec(50, 2) // 0.50
+	s.stakingKeeper.EXPECT().StakingTokenSupply(s.ctx).Return(stakingSupply, nil).Times(1)
+	s.stakingKeeper.EXPECT().BondedRatio(s.ctx).Return(bondedRatio, nil).Times(1)
+
+	expectedCoin := sdk.NewCoin("stake", math.NewInt(20))
+	expectedCoins := sdk.NewCoins(expectedCoin)
+
+	// Set bank keeper expectations for minting and fee collection.
+	s.bankKeeper.EXPECT().MintCoins(s.ctx, types.ModuleName, expectedCoins).Return(nil).Times(1)
+	s.bankKeeper.EXPECT().SendCoinsFromModuleToModule(s.ctx, types.ModuleName, authtypes.FeeCollectorName, expectedCoins).Return(nil).Times(1)
+
+	// Call the mint function.
+	err := s.mintKeeper.MintFn(s.ctx)
+	s.Require().NoError(err)
+
+	// Retrieve the updated minter from storage.
+	storedMinter, err := s.mintKeeper.Minter.Get(s.ctx)
+	s.Require().NoError(err)
+
+	s.Require().Equal(math.LegacyMustNewDecFromStr("0.130000005226169707"), storedMinter.Inflation)
+
+	// The dummy minter's NextAnnualProvisions returns 100.
+	s.Require().Equal(math.LegacyMustNewDecFromStr("130000005.226169707000000000"), storedMinter.AnnualProvisions)
```

Comment:
can you explain these values? in case this test breaks in the future, it would be nice to know where i should be looking and how this was calculated.

Reply:
[update](https://github.com/cosmos/cosmos-sdk/pull/24436/commits/0f783f65789c35096cdc0c316ce3a2615bfc7969)